src/Wishlist/Wishlist-button.jsx

```diff
@@ -0,0 +1,96 @@
+import { checkStatus, isAlreadyAdded } from "../array-update-functions";
+import { useStateContext } from "../context";
+import { serverRequest } from "../server-request";
+import { useEffect, useRef } from "react";
+
+export const WishlistButton = ({
+  product,
+  setMessage,
+  disableButtonWhileProcessing,
+  setDisableButton
+}) => {
+  const { state, dispatch } = useStateContext();
+
+  let isRendered = useRef(false);
+
+  useEffect(() => {
+    isRendered.current = true;
+    return () => {
+      isRendered.current = false;
+    };
+  }, []);
+
+  const addWishlistHandler = async () => {
+    setMessage({ msg: "adding to wishlist..", msgType: "toast-inform" });
+
+    setDisableButton(true);
+
+    try {
+      if (isAlreadyAdded(state.itemsInWishlist, product.id)) {
+        if (!checkStatus(state.itemsInWishlist, product.id)) {
+          await serverRequest({
+            requestType: "PUT",
+            url: "api/wishlists",
+            dataToOperateId: product.id,
+            dataToOperate: {
+              wishlist: {
+                ...product,
+                status: { exists: true }
+              }
```

Comment:
Why not POST request here ?

Reply:
I am trying to update the existing object on the server, hence PUT request